cmake/thirdparty/SetupMPI.cmake

```diff
@@ -44,7 +44,10 @@
 # MPI
 ################################
 
-find_package(MPI REQUIRED)
+if (ENABLE_FIND_MPI)
+    find_package(MPI REQUIRED)
+endif()
+
 message(STATUS "MPI C Compile Flags:  ${MPI_C_COMPILE_FLAGS}")
```

Comment:
Should the compile, include, link, and libraries message lines be in the if guard as well?